sdk/messaging/azwebpubsub/client_custom.go

```diff
@@ -0,0 +1,194 @@
+//go:build go1.18
+// +build go1.18
+
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package azwebpubsub
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/url"
+	"strings"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/runtime"
+	"github.com/Azure/azure-sdk-for-go/sdk/messaging/azwebpubsub/internal"
+	"github.com/golang-jwt/jwt"
+)
+
+const defaultExpirationTime = time.Hour
+
+// ClientOptions contains optional settings for [Client]
+type ClientOptions struct {
+	azcore.ClientOptions
+}
+
+// NewClient creates a client that manages Web PubSub service
+func NewClient(endpoint string, hub string, credential azcore.TokenCredential, options *ClientOptions) (*Client, error) {
+	if hub == "" {
+		return nil, errors.New("empty hub name is not allowed")
+	}
+
+	if options == nil {
+		options = &ClientOptions{}
+	}
+
+	authPolicy := runtime.NewBearerTokenPolicy(credential, []string{internal.TokenScope}, nil)
+	azcoreClient, err := azcore.NewClient(internal.ModuleName+".Client", internal.ModuleVersion,
```

Comment:
Per `azcore@v1.9.0` we now just want the module name as the first parameter, no `.Client` suffix.